kong/cmd/utils/prefix_handler.lua

```diff
@@ -828,7 +828,8 @@ local function prepare_prefix(kong_config, nginx_custom_template_path, skip_writ
   end
 
   if kong_config.admin_gui_listeners then
-    prepare_prefixed_interface_dir("/usr/local/kong", "gui", kong_config)
+    -- use LIBRARY_PREFIX as an alternative to /usr/local/kong in the Bazel dev environment
+    prepare_prefixed_interface_dir(os.getenv("LIBRARY_PREFIX") or "/usr/local/kong", "gui", kong_config)
```

Comment:
I'm slightly worried about the use of `LIBRARY_PREFIX` here as it's currently only used by dev scripts but now it could has effect in production as well. Some idea to improve it:
- Prefix it with `KONG_` to avoid surprises. We will then need to set the new env var in our dev script too.
- Use a hardcoded path when a env var indicating we are in dev env, instead ousing directly from the value of LIBRARY_PREFIX. This could help us prevent path traversal even if malicious user injected this env var in production.

Reply:
@fffonion I agree with you. I'm converting this PR into a draft and refactor it. Meanwhile, do we have best practices (e.g., env vars, helpers) to tell if we are in the dev env?

Reply:
We can use `BUILD_NAME` and use that to construct path. Or it just occurs to me that we can also produce a lua file to populate the prefix, and require that from the file that you change in this PR. So you don't need to rely on the 
fact that `BUILD_NAME` or some other vars are populated, which is done by the venv scripts/may not work in other tooling.

It can be a simple genrule that produce a lua file, something like
```starlark
genrule(
    name = "kong_path_config",
    outs = ["path/to/the/kong_path_config.lua"],
    local = 1,
    cmd = "echo 'return \"%s/kong\"' >$@" %  KONG_VAR["INSTALL_DESTDIR"],
)
```
then in lua we can just
```lua
local the_path = require "path.to.the.kong_path_config"
```

Reply:
Or to take a completely different path, that to start using of `datafile` to ship admin_gui assets, like the `saml` plugin did for `xml` assets would also be a good idea.

Reply:
Thanks for the suggestion. I've appended the `echo` command to the `kong` target which would be more straightforward.